tools/scripts/generate-sponsors.mts

```diff
@@ -2,102 +2,158 @@ import * as fs from 'node:fs';
+    for (const item of arr) {
+      (grouped[fn(item)] ??= []).push(item);
+    }
+    return grouped;
+  };
+  return Object.entries(
+    groupBy(
+      data.collective.members.nodes,
+      ({ account }) => account.name || account.id,
+    ),
+  ).flatMap(([id, members]) => {
+    const [
+      {
+        account: { website, ...account },
+      },
+    ] = members;
+    return website
+      ? {
+          id,
+          image: account.imageUrl,
+          name: account.name,
+          totalDonations: members.reduce(
+            (sum, { totalDonations }) => sum + totalDonations.valueInCents,
+            0,
+          ),
+          website,
+        }
+      : [];
+  });
+});
 
-const sponsors = Object.entries(
-  // TODO: use Object.groupBy in Node 22
-  members.nodes.reduce<Record<string, MemberNodes>>((membersById, member) => {
-    const { account } = member;
-    (membersById[account.name || account.id] ??= []).push(member);
-    return membersById;
-  }, {}),
+const thanksDevSponsorsPromise = jsonApiFetch<
+  Record<'dependers' | 'donors', ['gh' | 'gl', string, number][]>
+>('thanks.dev/v1/vip/dependee/gh/typescript-eslint').then(async ({ donors }) =>
+  (
+    await Promise.all(
+      donors
+        /* GitLab does not have an API to get a user's profile. At the time of writing, only 13% of donors
+             from thanks.dev came from GitLab rather than GitHub, and none of them met the contribution
+             threshold. */
+        .filter(([site]) => site === 'gh')
+        .map(async ([, id, totalDonations]) => {
+          const { name, ...github } = await jsonApiFetch<
+            Record<'avatar_url' | 'blog', string> & {
+              name: string | null;
+            }
+          >(`github.com/users/${id}`);
+          return name
+            ? {
+                id,
+                image: github.avatar_url,
+                name,
+                totalDonations,
+                website: github.blog || `https://github.com/${id}`,
+              }
+            : [];
+        }),
+    )
+  ).flat(),
+);
+
+const sponsors = (
+  await Promise.all<SponsorData[]>([
+    openCollectiveSponsorsPromise,
+    thanksDevSponsorsPromise,
+  ])
 )
-  .map(([id, members]) => {
-    const [{ account }] = members;
-    return {
-      id,
-      image: account.imageUrl,
-      name: account.name,
-      totalDonations: members.reduce(
-        (sum, { totalDonations }) => sum + totalDonations.valueInCents,
-        0,
-      ),
-      website: account.website,
-    };
-  })
+  .flat()
   .filter(
-    ({ id, name, totalDonations, website }) =>
+    ({ id, name, totalDonations }) =>
       !(
         filteredTerms.some(filteredTerm =>
           name.toLowerCase().includes(filteredTerm),
         ) ||
         excludedNames.has(id) ||
-        totalDonations < 10000 ||
-        !website
+        totalDonations < 10000
       ),
   )
   .sort((a, b) => b.totalDonations - a.totalDonations);
```

Comment:
[Style] Nitty nit: unnecessary line removal, and I liked the blank line for readability 😄 

```suggestion
  .sort((a, b) => b.totalDonations - a.totalDonations);

```